src/components/side-panel/layer-panel/layer-column-config.js

```diff
@@ -35,51 +36,68 @@ const TopRow = styled.div`
 
 export default class LayerColumnConfig extends Component {
   static propTypes = {
-    layer: PropTypes.object.isRequired,
+    columns: PropTypes.object.isRequired,
     fields: PropTypes.arrayOf(PropTypes.any).isRequired,
+    assignColumnPairs: PropTypes.func.isRequired,
+    assignColumn: PropTypes.func.isRequired,
     updateLayerConfig: PropTypes.func.isRequired,
-    fieldPairs: PropTypes.arrayOf(PropTypes.any)
+    columnPairs: PropTypes.object,
+    fieldPairs: PropTypes.arrayOf(PropTypes.any),
+    columnLabels: PropTypes.object
   };
 
+  columnPairs = props => props.columnPairs;
+  fieldPairs = props => props.fieldPairs;
+  fieldPairsSelector = createSelector(
+    this.columnPairs,
+    this.fieldPairs,
+    (columnPairs, fieldPairs) => columnPairs ?
+      fieldPairs.map(fp => ({
+          name: fp.defaultName,
+          type: 'point',
+          pair: fp.pair
+        }))
+      : null
+  );
+
   _updateColumn(key, value) {
-    const {layer} = this.props;
+    const {columnPairs, assignColumnPairs, assignColumn} = this.props;
 
     const columns =
-      value && value.pair && layer.columnPairs
-        ? layer.assignColumnPairs(key, value.pair)
-        : layer.assignColumn(key, value);
+      value && value.pair && columnPairs
+        ? assignColumnPairs(key, value.pair)
+        : assignColumn(key, value);
 
     this.props.updateLayerConfig({columns});
   }
 
   render() {
-    const {layer, fields, fieldPairs} = this.props;
+    const {
+      columns,
+      columnLabels,
+      fields,
+    } = this.props;
+
+    const fieldPairs = this.fieldPairsSelector(this.props);
+
     return (
       <div>
         <SidePanelSection>
           <div className="layer-config__column">
-          <TopRow>
-            <PanelLabel>Columns</PanelLabel>
-            <PanelLabel>* Required</PanelLabel>
-          </TopRow>
-          {Object.keys(layer.config.columns).map(key => (
-            <ColumnSelector
-              column={layer.config.columns[key]}
-              label={key}
-              key={key}
-              allFields={fields}
-              fieldPairs={
-                layer.columnPairs
-                  ? fieldPairs.map(fp => ({
-                      name: fp.defaultName,
-                      type: 'point',
-                      pair: fp.pair
-                    }))
-                  : null
-              }
-              onSelect={val => this._updateColumn(key, val)}
-            />
-          ))}
+            <TopRow>
+              <PanelLabel>Columns</PanelLabel>
+              <PanelLabel>* Required</PanelLabel>
+            </TopRow>
+            {Object.keys(columns).map(key => (
+              <ColumnSelector
+                column={columns[key]}
+                label={columnLabels && columnLabels[key] || key}
+                key={key}
+                allFields={fields}
+                fieldPairs={fieldPairs}
+                onSelect={val => this._updateColumn(key, val)}
```

Comment:
an possible optimization for this is to create a class method because it will avoit creating an anonymous function every time the component  renders